src/components/dashboard/Overview/Overview.tsx

```diff
@@ -47,7 +48,8 @@ const Overview = (): ReactElement => {
   const router = useRouter()
   const theme = useTheme()
   const isSmallScreen = useMediaQuery(theme.breakpoints.down('sm'))
-  const isSwapFeatureEnabled = useHasFeature(FEATURES.NATIVE_SWAPS)
+  const isCounterfactualSafe = useIsCounterfactualSafe()
+  const isSwapFeatureEnabled = useHasFeature(FEATURES.NATIVE_SWAPS) && !isCounterfactualSafe
```

Comment:
The overview buttons are already hidden for counterfactual safes so this is not necessary.